system/lib/wasmfs/proxied_async_js_impl_backend.h

```diff
@@ -108,6 +123,22 @@ class ProxiedAsyncJSImplFile : public DataFile {
 
   void flush() override {}
 
+  std::string getRelativeFileName() {
+    std::string name;
+    auto parent = locked().getParent();
+    auto child = shared_from_this();
+    while (parent) {
+      if (!name.empty())
+        name.insert(0, "/");
+      name.insert(0, parent->locked().getName(child));
+      if (parent->getBackend() != backend)
+        break;
+      child = parent;
+      parent = parent->locked().getParent();
+    }
+    return name;
+  }
```

Comment:
Reading the parent pointers and constructing file paths breaks the backend abstraction. If we need to keep track of objects across the C++/JS boundary, we should use integer indices into JS arrays rather than strings.

Reply:
But what could be a better way to construct a relative path to the file? I need it to be a part of the URL. I need to somehow obtain names of the path components and make them available to JS.

Reply:
Check out `node_backend.cpp`, which has the same problem of needing to track relative paths.

Reply:
Done. Used a similar approach as in `node_backend.cpp`. Now each backend object knows its name.